rest-folder.go

```diff
@@ -0,0 +1,145 @@
+package sdk
+
+/*
+   Copyright 2016 Alexander I.Grafov <grafov@gmail.com>
+   Copyright 2016-2019 The Grafana SDK authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+	   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+   ॐ तारे तुत्तारे तुरे स्व
+*/
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// GetAllFolders gets all folders.
+// Reflects GET /api/folders API call.
+func (r *Client) GetAllFolders() ([]Folder, error) {
```

Comment:
This is missing the pagination parameters. I'd suggest taking a look at `rest-annotation.go` and copying/pasting roughly the same thing here. It seems like there is a parameter called `limit` which we need to support.

Reply:
got this

Reply:
when I add this param support the ci not pass, and I don't see support for this parameter from the official documentation. my test passed locally, grafana version : v6.5.3, but I'm not sure if all versions support

Reply:
https://grafana.com/docs/grafana/v6.5/http_api/folder/
> GET /api/folders?limit=10 HTTP/1.1

It's weird that there is no parameter like `start` but :man_shrugging: 
It is definitely there: https://github.com/grafana/grafana/blob/8fd153edb75fa0bf8d74484718a2a36b14d8767a/pkg/api/folder.go#L15

Could you add it, please?

Reply:
Ah, I see that you have added it. Thanks a lot!